Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/viewholder/SwitchSettingViewHolder.java

```diff
@@ -33,10 +40,41 @@ public void bind(SettingsItem item)
 
     mBinding.settingSwitch.setChecked(mItem.isChecked(getAdapter().getSettings()));
 
-    mBinding.settingSwitch.setEnabled(mItem.isEditable());
+    // Check for IPL to make sure user can skip.
+    if (mItem.getSetting() == BooleanSetting.MAIN_SKIP_IPL)
+    {
+      ArrayList<String> iplDirs = new ArrayList<>(Arrays.asList("USA", "JAP", "EUR"));
+      for (String dir : iplDirs)
+      {
+        File iplFile = new File(DirectoryInitialization.getUserDirectory(),
+                File.separator + "GC" + File.separator + dir + File.separator + "IPL.bin");
+        if (iplFile.exists())
+        {
+          iplExists = true;
+          break;
+        }
+      }
+
+      if (mItem.isChecked(getAdapter().getSettings()))
+      {
+        mBinding.settingSwitch.setEnabled(iplExists);
+      }
+      else
+      {
+        mBinding.settingSwitch.setEnabled(true);
```

Comment:
You should also call `setEnabled(true)` when the setting isn't the IPL setting, otherwise the switch may incorrectly show up as disabled if the view gets recycled.

Reply:
That won't be a problem and is already set correctly by 
`mBinding.settingSwitch.setChecked(mItem.isChecked(getAdapter().getSettings()));`
on line 42.

Reply:
Do you mean that `setChecked` implicitly has the effect of `setEnabled(true)`?

Reply:
Oh shoot, I misread that badly. I'll add `setEnabled(true)` after `setChecked(...)` so we don't have a problem with recycling. Sorry about that.